test/performance/README.md

```diff
@@ -0,0 +1,38 @@
+# Performance tests
+
+This directory contains performance tests for `Knative Serving`:
+
+- Tests are using [kperf](https://github.com/knative-sandbox/kperf); a testing tool designed to run load tests in knative
+
+## Prerequisites
+
+- If running the tests using `go test` then you need to install kperf since the tests assumes its already installed.
+- Running `performance-tests.sh` installs kperf in the temporary folder to enable the ci to run the tests.
+- to install kperf:
+  - run `git clone https://github.com/knative-sandbox/kperf.git`
+  - run `make build`
+
+## Flags
+
+- These flags are used by the performance tests framework used here:
+  - `service-prefix`: The prefix for the services generated by kperf.
+  - `services-count` The count of Services generated by kperf.
+  - `kperf-output`: kperf output file(s) location.
+  - `service-average`: The average time tests should expect for the kperf service creation.
+  - `service-timeout`: The time the tests should wait for the kperf Service creation timeout.
+
+- Flags required for accessing GCS bucket to either upload results, or compare with previously upload results:
+  - `service-account`: Location of service account file for GCP bucket access.
+  - `bucket`: GCS bucket name.
+  - `combine-results`: Set this flag to generate combined resutls of current run with reference latest run.
+  - `upload-results`: Set this flag to upload latest resutls of current to reference bucket.
+
+## Running performance tests
+
+To run the tests, you need to have a running environment that meets
+[the e2e test environment requirements](#environment-requirements), and you need
+to specify the build tag `performance`.
```

Comment:
Just started working my way through this (sorry I wasn't able to get to it sooner!)...

Does `TestPerformanceScaleFromZero` require Istio for the networking layer? I usually use Kourier locally when running the e2e tests, but the test failed for me with this error: `result of scale is error: failed to get the cluster endpoint: services "istio-ingressgateway" not found`. 

The test passes when using Istio, so assuming we should call out that dependency here since the [e2e docs](https://github.com/knative/serving/blob/main/DEVELOPMENT.md#deploy-knative-ingress) suggest using Kourier.

Reply:
it doesn't require istio, but read the env variables `GATEWAY_OVERRIDE` and `GATEWAY_NAMESPACE_OVERRIDE` 

I'm open to better ideas about this.